### ISSUE TITLE
Can't stream data

### ISSUE THREAD

**Issue by Tsar** [kai-morich/SimpleBluetoothLeTerminal#24]
Hello.

Thank you for this BLE usage sample!

I am using JDY-10 UART-BluetoothLE module. I ran into following problem: everything larger than 56 bytes that I send through JDY-10 at baud rate 115200 is trunkated to 56 bytes (Android receives it as 3 `onCharacteristicChanged`: 20+20+16), only the beginning of packet is received.
When I try using small packets (10-11 bytes each) and send them almost at maximum speed of UART, something around 70% of data is lost.

Is that OK (BLE is just not capable to send faster)? Or am I doing something wrong?
I read that BLE supports 1 Mbits/sec, seems a lot more than 115200.

P.S. I found that `onMtuChanged` logs mtu size 23, payload size 20. Did not become 512.

**Comment by kai-morich** [comment]
Hi,

the MTU size is typically limited by the bluetooth device and typically only devices > BLE 4.0 support larger sizes.
Maybe your BLE device has a limited buffer size, so data is truncated after 56 bytes.
[Here](https://interrupt.memfault.com/blog/ble-throughput-primer) you can see that the effictive throughput is ~ 0.2 Mbit/sec

**Comment by Tsar** [comment]
Thank you!

Seems you're right, and it's BLE 4.0. I can't set higher MTU size than 23.
I suppose that even more speed loss happens when the device converts from UART to BLE, does it's own buffering or something.

I tried sending 5000 bytes during 0.5 sec (500 ms) and playing with size of UART packets (I mean not actual packets, but size of buffer sent at once to UART). The best result I was able to achieve was receiving 4040 bytes on Android, but it's not stable (same settings sometimes allow me to receive only 2320 bytes of 5000).

For comparison, UART-to-USB device allows me to stream 11000 bytes per second and receive them without any loss. UART packet size does not matter.

Seems that I need other BLE device. Or Bluetooth 2.0 maybe.
